### DISCUSSION TITLE
Conversion from float to str seems to go wrong

### DISCUSSION THREAD

**Discussion by lverweijen** [sqlalchemy/sqlalchemy#11891] [Usage Questions]
I think this is a bug, but I was told to start a discussion instead.

Python version: 3.11.9
SqlAlchemy version: 2.0.26
Backend: Sql Server (through pyodcb, using driver "ODBC driver 13/18 for Sql Server"). I tried both.

Given a model such as:

```python
Base = sqlalchemy.declarative_base()

class Table(Base):
    __table__ = "mytable"
    str_field = db.Column(db.String(1000), nullable=True)
```

Then do something like:
```py
# Sql Server can be downloaded for free
# Set up db in a suitable way
# db.metadata.create_all()

t = Table()
t.str_field = 0.001

with db.session.begin():
    db.session.add(t)

db.session.commit(t)
```

No error is shown, but looking in the database, it set str_field to "1.00000000" instead of the expected "0.001".

It can be avoided by doing:
```py
t.str_field = str(0.001)
```

This might be related to:
https://github.com/sqlalchemy/sqlalchemy/discussions/6365

But I don't know.

I would expect a good conversion or a warning that it doesn't convert.
This might also be an error in the driver for Sql Server.
 I'm sorry if this example isn't minimal enough to run on itself. Furthermore, I don't know if this happens in other database backends.

**Comment by zzzeek** [comment]
SQLAlchemy doesn't validate and/or coerce values for most datatypes and the conversion you are seeing happens entirely within pyodbc, so if you think that conversion is a bug then you'd want to post on their list.

if you want a string that does validation and coercion of data before it's passed to the database see examples such as https://docs.sqlalchemy.org/en/20/core/custom_types.html#coercing-encoded-strings-to-unicode

**Comment by gordthompson** [comment]
https://github.com/mkleehammer/pyodbc/issues/753
